test/org/pentaho/di/trans/steps/mapping/MappingTest.java

```diff
@@ -250,4 +250,55 @@ public void testMapping_WhenSharingPreviousStepWithAnother() throws Exception {
 
     assertEquals( 0, trans.getErrors() );
   }
+
+
+  /**
+   * This test case relates to PDI-13545. It executes a transformation with a Mapping step that is not configured
+   * manually with an <tt>outputStepname</tt> property and, therefore, it is set to be a mapping output step from the
+   * internal transformation.
+   *
+   * @throws Exception
+   */
+  public void testMapping_WhenNextStepHasTwoCopies_AndOutputIsNotDefinedExplicitly() throws Exception {
+    runTransWhenMappingsIsFollowedByCopiedStep(
+      "testfiles/org/pentaho/di/trans/steps/mapping/pdi-13545/PDI-13545-1.ktr" );
+  }
+
+  /**
+   * This test case relates to PDI-13545. It executes a transformation with a Mapping step that is configured manually
+   * with an <tt>outputStepname</tt> property.
+   *
+   * @throws Exception
+   */
+  public void testMapping_WhenNextStepHasTwoCopies_AndOutputIsDefinedExplicitly() throws Exception {
+    runTransWhenMappingsIsFollowedByCopiedStep(
+      "testfiles/org/pentaho/di/trans/steps/mapping/pdi-13545/PDI-13545-2.ktr" );
```

Comment:
@mattyb149, have fixed it already: https://github.com/pentaho/pentaho-kettle/pull/1095/